src/core/temporal/qgsabstracttemporal.h

```diff
@@ -0,0 +1,125 @@
+/***************************************************************************
+                         qgsabstracttemporal.h
+                         ---------------
+    begin                : January 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef QGSABSTRACTTEMPORAL_H
+#define QGSABSTRACTTEMPORAL_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgslogger.h"
+#include "qgsmessagelog.h"
+#include "qgsapplication.h"
+#include "qgsrange.h"
+
+#include <QObject>
+#include <QDateTime>
+
+/**
+ * \class QgsAbstractTemporal
+ * \ingroup core
+ * Base class for temporal based classes.
+ *
+ * Subclasses may wish to update the abstract temporal functions.
+ *
+ * \since QGIS 3.14
+ */
+class CORE_EXPORT QgsAbstractTemporal
```

Comment:
```suggestion
class CORE_EXPORT QgsAbstractTemporal
```

QGIS convention is to name classes "abstract" only when they have at least one pure virtual method. `QgsTemporalRangeObject` might be a better name.